### PR TITLE
Add survey code

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: app/lib/templates.dart [hunk 1]
```diff
@@ -300,14 +302,13 @@ class TemplateService {
...
-    String faviconUrl,
-    String searchQuery,
-  }) {
+  String renderLayoutPage(String title, String contentString,
+      {PackageVersion packageVersion,
+      Map<String, String> pageMapAttributes,
+      String faviconUrl,
+      String searchQuery,
+      bool includeSurvey: true}) {
+    includeSurvey ??= true;
```
Comment (mkustermann) [inline app/lib/templates.dart]: It makes little sense to have `bool includeSurvey: true` if you check again for `null` and set it to `true`.

Either the caller is responsible for never passing `includeSuvey: null` or you remove the `: true`.

File: app/lib/templates.dart [hunk 2]
```diff
@@ -384,24 +386,21 @@ class TemplateService {
...
-    String searchQuery,
-  }) {
+  String _renderPage(String template, values,
+      {String title: 'pub.dartlang.org',
+      PackageVersion packageVersion,
+      Map<String, String> pageMapAttributes,
+      String faviconUrl,
+      String searchQuery,
+      bool includeSurvey: true}) {
+    includeSurvey ??= true;
```
Comment (mkustermann) [inline app/lib/templates.dart]: ditto

File: app/views/layout.mustache [hunk 3]
```diff
@@ -45,6 +45,9 @@
     </PageMap>
     -->
     {{/has_pagemap}}
+    {{#include_survey}}
+    <script async="" defer="" src="//www.google.com/insights/consumersurveys/async_survey?site=5wdvu4vc5lhop3xbatxof6lzcm"></script>
```
Comment (mkustermann) [inline app/views/layout.mustache]: Why would we want to link to the **http** version of `www.google.com` if the pub server is running on `http`?

Even if the pub server is running on **non-https** the link should still be **https://www.google.com/...** I think.

So consider changing `//` with `https://`.

**Comment by mkustermann** [comment]
Looks fine, two minor comments.

File: app/views/layout.mustache (hunk 3 above)
Comment (mkustermann) [inline app/views/layout.mustache]: Or is this not allowed for some reason?

Comment (kevmoo) [inline app/views/layout.mustache]: This is copy-pasted the recommended usage. Given that we host on https, I'm going to leave as-is.

File: app/lib/templates.dart (hunk 1 above)
Comment (kevmoo) [inline app/lib/templates.dart]: ack

**Pull request merged**